airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -980,8 +982,16 @@ def execute(self, context):
                 project_id=self.project_id, job=self.job["job"], location=self.region,
             )
             job_id = job_object.reference.job_id
-            self.hook.wait_for_job(job_id=job_id, location=self.region, project_id=self.project_id)
-            self.log.info('Job executed correctly.')
+            if not self.asynchronous:
+                self.hook.wait_for_job(
+                    job_id=job_id,
+                    location=self.region,
+                    project_id=self.project_id
+                )
+                self.log.info('Job executed correctly.')
```

Comment:
nit: this language is vague. Also IMO we can log job id so someone inspecting the logs does not also have to look at XCom return value.
```suggestion
                self.log.info(f'Job {job_id} completed successfully.')
```